os/arch/arm/src/armv8-m/up_initialstate.c

```diff
@@ -114,6 +114,15 @@ void up_initial_state(struct tcb_s *tcb)
 	/* Set the stack limit value */
 
 	xcp->regs[REG_R10] = (uint32_t)tcb->stack_alloc_ptr + 64;
+#endif
+#ifdef CONFIG_REG_STACK_OVERFLOW_PROTECTION
+	xcp->regs[REG_SPLIM]   = (uint32_t)tcb->stack_alloc_ptr;
+#endif
+#ifdef CONFIG_ARMV8M_TRUSTZONE
```

Comment:
This is related to Trustzone and has been removed in a different PR which we raised for fixing TZ context issues. 
Please remove this change as its not related to SP LIM register.

Reply:
modified